crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -231,7 +232,7 @@ where
         + BlockIdReader
         + CanonChainTracker
         + StageCheckpointReader
-        + ChainSpecProvider
+        + ChainSpecProvider<ChainSpec = ChainSpec>
```

Comment:
nice, setting the AT to a concrete type makes it possible to impl this in incremental prs.